picard/browser/browser.py

```diff
@@ -166,11 +174,17 @@ def do_GET(self):
             log.error('Browser integration failed handling request', exc_info=True)
             self._response(500, 'Unexpected request error')
 
+    def _fmt_message(self, format, args):
+        message = format % args
+        return "%s %s" % (self.address_string(), message.translate(self._control_char_table))
+
     def log_error(self, format, *args):
-        log.error(format, *args)
+        message = self._fmt_message(format, args)
+        log.error(message)
```

Comment:
There is a minor change in behavior here. `log.error` only makes the argument substitution, if the message will get actually logged. Now the message gets processed regardless of log level.

Not sure, it's probably not really relevant in this context.

Reply:
Yes, but it reduces code redundancy, not much impact as it is just when queries are handled, unlikely critical.

Reply:
See https://github.com/metabrainz/picard/commit/31945b31ac8e6360763a369dd05e443970a10a87